org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandlerTest.java

```diff
@@ -1176,6 +1177,72 @@ public void testCompletion_testClassesAvailableIntoTestCode() throws Exception {
 		assertEquals("AbstractTest - foo.bar", list.getItems().get(0).getLabel());
 	}
 
+	@Test
+	public void testStaticImports1() throws Exception {
+		PreferenceManager.getPrefs(null).setJavaContentAssistFavoriteMembers("test1.A.foo");
+		try {
+			ICompilationUnit unit = getWorkingCopy("src/test1/B.java",
+			//@formatter:off
+				"package test1;\n" +
+				"\n" +
+				"public class B {\n" +
+				"    public void bar() {\n" +
+				"        f\n" +
+				"    }\n" +
+				"}\n");
+			//@formatter:on
+
+			int[] loc = findCompletionLocation(unit, "f");
+			CompletionList list = server.completion(JsonMessageHelper.getParams(createCompletionRequest(unit, loc[0], loc[1]))).join().getRight();
+			assertNotNull(list);
+			assertTrue(list.getItems().size() > 0);
+			CompletionItem item = null;
+			for (CompletionItem it: list.getItems()) {
+				if ("foo() : void".equals(it.getLabel())) {
+					item = it;
+					break;
+				}
+			}
+			assertNotNull("no proposal for foo()", item);
+		} finally {
+			PreferenceManager.getPrefs(null).setJavaContentAssistFavoriteMembers("");
+		}
+	}
+
+	@Test
+	public void testStaticImports2() throws Exception {
+		PreferenceManager.getPrefs(null).setJavaContentAssistFavoriteMembers("test1.A.foo");
+		try {
+			ICompilationUnit unit = getWorkingCopy("src/test1/B.java",
+			//@formatter:off
+					"package test1;\n" +
+					"\n" +
+					"public class B {\n" +
+					"    public void bar() {\n" +
+					"        /* */f\n" +
+					"    }\n" +
+					"    public void foo(int x) {\n" + // conflicting method, no static import possible
+					"    }\n" +
+					"}\n");
+				//@formatter:on
+
+			int[] loc = findCompletionLocation(unit, "/* */f");
+			CompletionList list = server.completion(JsonMessageHelper.getParams(createCompletionRequest(unit, loc[0], loc[1]))).join().getRight();
+			assertNotNull(list);
+			assertTrue(list.getItems().size() > 0);
+			CompletionItem item = null;
+			for (CompletionItem it : list.getItems()) {
+				if ("foo() : void".equals(it.getLabel())) {
+					item = it;
+					break;
+				}
+			}
+			assertNotNull("there is a proposal for foo()", item);
```

Comment:
shouldn't it be assertNull()? since foo() is not supposed to be imported. If yes, then assertion can  be made directly in loop, instead of breaking it